### DISCUSSION TITLE
Suggestion to rename `token` to `asset`

### DISCUSSION THREAD

**Discussion by razgraf** [sablier-labs/evm-monorepo#273] [General]
Should we consider renaming the streamed `token` into a streamed `asset`?

In v2, the stream itself is a token (an NFT to be specific). As it inherits attributes and methods from ERC721, it gets a `tokenId` and other "token"-related features. This could make it a bit unclear for the underlying ERC20 the stream runs on. There will be getters for things like `stream.token, stream.tokenId` which are very different things but due to naming will most often live alongside each other.

@PaulRBerg @andreivladbrg

**Comment by PaulRBerg** [comment]
As I have argued on Slack, I am in favor of this proposal.

1. As you say, renaming `token` to `asset` would create a distinction between the NFT token and the ERC-20 token (I had to wrestle with this issue when I implemented #249 and ended up naming the function `getERC20Token`, which I still find ugly)
2. We are now calling Sablier a "cryptoasset streaming protocol" - here's your `asset`!
3. Another benefit of using `asset` is that it opens up the "use case space" for what can be streamed as long as that contract is compliant with the ERC-20 standard. Maybe the contract developer prefers to refer to the contract as a "shares contract", or it's some NFT fractionalization scheme. In these cases, "asset" fits better as a description because it's more generic.

**Reply by PaulRBerg** [reply]
Created an issue to track the implementation for this:

https://github.com/sablierhq/v2-core/issues/274

**Comment by PaulRBerg** [comment]
Locking since this was implemented in #275.
